bindings/python/shellcode.py

```diff
@@ -93,11 +144,15 @@ def test_i386(mode, code):
         # tracing all instructions with customized callback
         mu.hook_add(UC_HOOK_CODE, hook_code)
 
-        # handle interrupt ourself
-        mu.hook_add(UC_HOOK_INTR, hook_intr)
-
-        # handle SYSCALL
-        mu.hook_add(UC_HOOK_INSN, hook_syscall, None, 1, 0, UC_X86_INS_SYSCALL)
+        if mode == UC_MODE_32:
+            # handle interrupt ourself
+            mu.hook_add(UC_HOOK_INTR, hook_intr)
+        
+        if mode == UC_MODE_32:
```

Comment:
this is in the same block `if mode == UC_MODE_32`, so you can remove this line, and the blank line above